apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/GRPCChannelManager.java

```diff
@@ -91,6 +97,23 @@ public void shutdown() {
 
     @Override
     public void run() {
+        if (DNS_PERIOD_RESOLVE_ACTIVE) {
+            String backendService = Config.Collector.BACKEND_SERVICE.split(",")[0];
+            try {
+                String[] domainAndPort = backendService.split(":");
+
+                List<String> newGrpcServers = Arrays
+                        .stream(InetAddress.getAllByName(domainAndPort[0]))
+                        .map(InetAddress::getHostAddress)
+                        .map(ip -> String.format("%s:%s", ip, domainAndPort[1]))
+                        .collect(Collectors.toList());
+
+                grpcServers = newGrpcServers;
+            } catch (Throwable t) {
+                LOGGER.error(t, "Failed to resolve {} of backend service.", backendService);
+            }
+        }
+
```

Comment:
I think this should be called only when gRPC connection available. `reconnect==true`

Reply:
fixed